ocp_on_osp.yml

```diff
@@ -1,309 +1,42 @@
-        master_etcd_nvme_config: "{{ lookup('template', 'master-etcd.ign.json.j2') }}"
-      when: ocp_master_etcd_nvme
-
-    - name: Dump new version of master.ign file
-      copy:
-        content: "{{ master_ign_config_base | combine(master_etcd_nvme_config) | to_json }}"
-        dest: "/home/stack/{{ ocp_cluster_name }}/{{ item }}"
-      with_items:
-        - master.ign
-        - master.ign.etcd
-      when: ocp_master_etcd_nvme
-
-    - name: Read original worker.ign config file
-      slurp:
-        src: "/home/stack/ocp_clusters/{{ ocp_cluster_name }}/worker.ign.orig"
-      register: worker_ign
-      when: worker_nvme_ephemeral
-
-    - name: Get worker nvme ephemeral storage ignition configuration
-      set_fact:
-        worker_ign_config_base: "{{ worker_ign['content'] | b64decode }}"
-        worker_nvme_config: "{{ lookup('template', 'worker-nvme.ign.json.j2') }}"
-      when: worker_nvme_ephemeral
-
-    - name: Dump new version of worker.ign file
-      copy:
-        content: "{{ worker_ign_config_base | combine(worker_nvme_config) | to_json }}"
-        dest: "/home/stack/ocp_clusters/{{ ocp_cluster_name }}/{{ item }}"
-      with_items:
-        - worker.ign
-        - worker.ign.nvme
-      when: worker_nvme_ephemeral
-
-    - name: run installer
-      shell: |
-        ./openshift-install --log-level=debug create cluster --dir={{ ocp_cluster_name }} > ocp_install.log 2>&1
-      args:
-        chdir: /home/stack/
-
-    - name: install ocp client
-      block:
-        - name: get the ocp client
-          get_url:
-            validate_certs: no
-            url: "{{ ocp_installer_url }}{{ ocp_release }}/openshift-client-linux.tar.gz"
-            dest: /home/stack/openshift-client-linux.tar.gz
-            force: yes
-
-        - name: untar ocp client
-          unarchive:
-            src: /home/stack/openshift-client-linux.tar.gz
-            dest: /home/stack/
-            remote_src: yes
-            mode: 0700
-
-        - name: copy oc to bin
-          shell: |
-            sudo cp /home/stack/oc /usr/local/bin
-            sudo chmod a+x /usr/local/bin/oc
-
-        - name: copy kubectl to bin
-          shell: |
-            sudo cp /home/stack/kubectl /usr/local/bin
-            sudo chmod a+x /usr/local/bin/kubectl
-      when: install_client == true
-
-    - name: create .kube dir on home
-      file:
-        path: /home/stack/.kube
-        state: directory
-        mode: a+rwx
-
-    - name: copy the kubeconfig file
-      copy:
-        src: /home/stack/{{ ocp_cluster_name }}/auth/kubeconfig
-        dest: /home/stack/.kube/config
-        mode: a+rx
-        remote_src: yes
-
-- import_playbook: create_ocp_infra_nodes.yml
-  when: (ocp_create_infra_nodes|bool == true or ocp_create_workload_nodes|bool == true)
+    - name: Add undercloud host to inventory for each cluster to be created
+      add_host:
+        name: "undercloud-{{ item }}"
+        group: underclouds
```

Comment:
Is there no better way to do this? Having a group called underclouds with the same host repeated multiple times doesn't seem optimal.

Reply:
> Is there no better way to do this? Having a group called underclouds with the same host repeated multiple times doesn't seem optimal.

I think this can be further optimized by overriding fork and serial values

Reply:
can't we just have group undercloud and use it for creating each cluster?

Reply:
> can't we just have group undercloud and use it for creating each cluster?

We have only one group called 'underclouds' under which same undercloud_host will be repeated. And this inventory is prepared dynamically.
This is needed when installing multiple openshift clusters.

Reply:
> Is there no better way to do this? Having a group called underclouds with the same host repeated multiple times doesn't seem optimal.

@sanjaychari 
The default forks value is 5, which means ansible maintains only 5 parallel ssh connections by default.
By overriding fork value to desired count, we can achieve parallelism.. which should be optimal as any other method.
This method also has an advantage when we want to install openshift clusters in batches.

Example usage:
ansible-playbook -vvv -f 25 ocp_on_osp.yml

Reply:
any update on this?

Reply:
As discussed @rajeshP524  will come up with better naming for group `underclouds` which is misleading

Reply:
sure @asyedham

Reply:
updated the name